storage/shard.go

```diff
@@ -108,7 +106,9 @@ type dbShard struct {
 }
 
 type dbShardRuntimeOptions struct {
-	writeNewSeriesAsync bool
+	writeNewSeriesAsync      bool
```

Comment:
Any reason to reuse the shard's lock instead of giving the struct a lock?

Reply:
There is a pretty good reason for this, so when we perform a write we get the read lock to check whether a series exists or not.  This requires just a single read lock, and to avoid needing to require multiple read locks we put this under the same read lock.

Reply:
Makes sense. Could you put that a NB blurb somewhere so we can capture for posterity

Reply:
Sure thing.